src/domain/core/src/testing/mock_dataset_repository.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use opendatafabric::{AccountName, DatasetHandle, DatasetRef};
+use url::Url;
+
+use crate::{
+    Dataset,
+    DatasetHandleStream,
+    DatasetRegistry,
+    DatasetRepository,
+    GetDatasetError,
+    GetDatasetUrlError,
+};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+mockall::mock! {
+    pub DatasetRepository {}
```

Comment:
How is this different from `automock` ?

Reply:
`automock` doesn't work in case we have super trait:
```rust
pub trait DatasetRepository: DatasetRegistry + Sync + Send {
#                            ^^^^^^^^^^^^^^^
```

Other options, I've also tried